android-app/app/src/main/java/de/dhbw/ka/se/fibo/ui/home/HomeFragment.java

```diff
@@ -51,6 +56,36 @@ public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceStat
         recyclerView.setLayoutManager(new LinearLayoutManager(view.getContext()));
         recyclerView.setAdapter(new ListAdapter(getContext(), ApplicationState.getInstance(requireContext()).getCashflows()));
         actionButton.setOnClickListener(e -> Navigation.findNavController(view).navigate(R.id.action_navigation_home_to_navigation_adding));
+
+        swipeRefreshLayout.setOnRefreshListener(() -> {
+            try {
+                Toast.makeText(requireContext(), R.string.list_refresh_in_progress, Toast.LENGTH_SHORT).show();
+
+                ApplicationState.getInstance(requireContext()).syncCashflows(result -> {
+                    swipeRefreshLayout.setRefreshing(false);
+
+                    ListAdapter adapter = (ListAdapter) recyclerView.getAdapter();
+
+                    requireActivity().runOnUiThread(() -> {
+                        if (null == adapter) {
+                            return;
+                        }
+
+                        if (!result.wasSuccessful()) {
+                            Toast.makeText(requireContext(), R.string.list_refresh_unsuccessful, Toast.LENGTH_LONG).show();
+                            return;
+                        }
+
+                        adapter.clear();
+                        adapter.addAll(new ArrayList<>(result.getCashflows()));
+                    });
+                });
+            } catch (IllegalStateException e) {
+                swipeRefreshLayout.setRefreshing(false);
+
+                Toast.makeText(requireContext(), R.string.list_refresh_during_illegal_state, Toast.LENGTH_LONG).show();
+            }
+        });
```

Comment:
there are no onSuccess and onError-Listener here because the Volley framework may result into errors, but is not the single source. The consumer of the synchronization should not be concerned with handling volley errors.

Non-critical errors can be detected via `result.wasSuccessful()` as well as `result.getThrowable()`.

Reply:
Alright i'm fine with it, could you maybe still extract this action listener into a method for a better code overview?

Reply:
sure thing!